### ISSUE TITLE
[cloud_firestore_web] startAfterDocument() skipping documents with the same value.

### ISSUE THREAD

**Issue by RedTech64** [FirebaseExtended/flutterfire#2936]
**Describe the bug**
`startAfterDocument()` is skipping documents with the same field value as it is sorting by. It seems to be acting just like `startAfter()`, where this behavior would be expected.

**To Reproduce**
1. Create a set of documents with a field where some of the values repeat.
2. Query the documents and sort them by the field. Limit the query to where the query would end in the middle of two documents with the same value.
4. Query the next set of documents using `startAfterDocument(lastDoc)` where `lastDoc `is the last document in the first query.

**Example**

I have 10 documents with a "quantity" field as shown below.
![Screenshot_205](https://user-images.githubusercontent.com/12768272/87234441-827d9380-c396-11ea-93aa-f4822a7c26d7.png)

Using the following code with two queries:
```dart
  CollectionReference colRef = Firestore.instance.collection('orders');
  QuerySnapshot docs = await colRef.orderBy('quantity').limit(5).getDocuments();
  print('First Query');
  docs.documents.forEach((element) => print(element.documentID));
  DocumentSnapshot lastDoc = docs.documents.last;
  QuerySnapshot moreDocs = await colRef.orderBy('quantity').startAfterDocument(lastDoc).limit(5).getDocuments();
  print('Second Query');
  moreDocs.documents.forEach((element) => print(element.documentID));
```

I would expect the output to be:
```
First Query
1004941
1004956
1004905
1004923
1004940
Second Query
1004947
1004953
1004957
1004958
1004903
```

(This is how it appear in Python)

But the output is:
```
First Query
1004941
1004956
1004905
1004923
1004940
Second Query
1004903
```

(This is how it appear in Flutter web)

**Expected behavior**
The second query should end exactly where the first query left off. BUT, it will skip all the documents with that value and go to the next.

**Additional context**
I did this exact same set up using the Python Admin SDK and it worked as I expected, but in Flutter it seems to skip values. I was not able to test this on Android or iOS, I was only able to test this on web.

**Flutter doctor**
FYI - I'm not using Android at all. Only using web.
```
Run `flutter doctor` and paste the output below:
[√] Flutter (Channel beta, 1.19.0-4.3.pre, on Microsoft Windows [Version 10.0.18363.836], locale en-US)
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions.
[√] Chrome - develop for the web
[√] Android Studio (version 3.5)
[√] VS Code (version 1.46.1)
[√] Connected device (2 available)

! Doctor found issues in 1 category.
```

**Comment by TahaTesser** [comment]
Hi @RedTech64 
Can you please provide your `flutter run --verbose` and a complete reproducible minimal code sample and screenshot firestore dummy data (firestore doesn't allow duplicate documents with sam ID) 
Thank you

**Comment by greghesp** [comment]
Hi @RedTech64 

I have tested this on the [current roadmap version](https://github.com/FirebaseExtended/flutterfire/issues/2582) like so:

```
      CollectionReference collectionReference = FirebaseFirestore.instance.collection('orders');
      QuerySnapshot docs = await collectionReference.orderBy('quantity').limit(5).get();
      print('First Query');

      docs.docs.forEach((element) {
        print(element.id);
      });

      DocumentSnapshot lastDoc = docs.docs.last;
      QuerySnapshot moreDocs = await collectionReference.orderBy('quantity').startAfterDocument(lastDoc).limit(5).get();

      print("Second Query");
      moreDocs.docs.forEach((element) {
        print(element.id);
      });
```

Everything appears to be functioning correctly and logs out as you expected:

```
First Query
1004941
1004956
1004905
1004923
1004940
Second Query
1004947
1004953
1004957
1004958
1004903
```

If the issue still exists on your current version, i'd recommend keeping an eye out on the roadmap for futher updates and releases.

**Comment by RedTech64** [comment]
@greghesp @TahaTesser 
Looks like this is fixed in the latest dev version (0.14.0-dev.1). It is certainly still a bug on the current version (0.13.7). I hope this can be rolled out soon.
